src/python/ensembl/io/genomio/manifest/manifest.py

```diff
@@ -0,0 +1,176 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Representation of a manifest file."""
+
+__all__ = ["Manifest", "ManifestError"]
+
+import hashlib
+import json
+import logging
+from pathlib import Path
+from typing import Any, TypeAlias
+
+
+ManifestDict: TypeAlias = dict[str, dict[str, Any]]
+
+
+class ManifestError(Exception):
+    """Could not load a manifest file."""
+
+
+class Manifest:
+    """Records of a manifest file and its files and md5 checksums."""
+
+    same_names = {
+        "gff3",
+        "fasta_dna",
+        "fasta_pep",
+        "functional_annotation",
+        "genome",
+        "seq_attrib",
+        "seq_region",
+        "agp",
+        "events",
+    }
+    alias_names = {
+        "gene_models": "gff3",
+        "dna": "fasta_dna",
+        "pep": "fasta_pep",
+    }
+    names = {name: name for name in same_names}
+    names = {**names, **alias_names}
```

Comment:
Not too important, but maybe worth "protecting" those attributes that users should not use:
```suggestion
    _same_names = {
        "gff3",
        "fasta_dna",
        "fasta_pep",
        "functional_annotation",
        "genome",
        "seq_attrib",
        "seq_region",
        "agp",
        "events",
    }
    _alias_names = {
        "gene_models": "gff3",
        "dna": "fasta_dna",
        "pep": "fasta_pep",
    }
    _same_names_dict = {name: name for name in _same_names}
    names = {**_same_names_dict, **alias_names}
```